src/BaseModel.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Friendica;
+
+use Friendica\Database\Database;
+use Friendica\Network\HTTPException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class BaseModel
+ *
+ * The Model classes inheriting from this abstract class are meant to represent a single database record.
+ * The associated table name has to be provided in the child class, and the table is expected to have a unique `id` field.
+ *
+ * @property int id
+ */
+abstract class BaseModel
```

Comment:
I guess I won't understand it anyhow :-)

Reply:
I'll try anyway: Object properties are normally accessed with `$object->property`. This is different from arrays where you use `$array['key']`. However, by implementing the `ArrayAccess` interface, you would be able to write `$object['property']` and it would work.

Is it something you'd prefer?

Reply:
Currently this is only done in the API, isn't it? Then I guess it hasn't got that high priority.

Reply:
I created the `Introduction` model class to be used across the codebase, not just for the API. Just tell me if you'd rather have a familiar way of using model objects as arrays, I'll add it to my to-do for the next release.

Reply:
The only limitation at the moment is that you can't set the values yet. The Model object I created has its attributes read-only, with verb methods to change its internal values.

So after I implement `ArrayAccess`, you will be able to do `$var = $object['property'];` but not `$object['property'] = $var;`.

Reply:
Maybe a  todo for a followup .. I'll save it in my backlog for further discussions ...